network/network.go

```diff
@@ -153,32 +152,23 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
 		}
 	}
 
-	var wg sync.WaitGroup
-
 	for _, host := range hosts {
-		wg.Add(1)
-
-		go func(host string) {
-			defer wg.Done()
+		start := time.Now()
 
-			start := time.Now()
+		conn, err := c.dialer.Dial("tcp", host)
+		if err != nil {
+			c.logger.Log("level", "error", "message", "could not dial host", "host", host, "stack", microerror.Stack(err))
+			c.dialErrorCount.WithLabelValues(host).Inc()
+			return
+		}
 
-			conn, err := c.dialer.Dial("tcp", host)
-			if err != nil {
-				c.logger.Log("level", "error", "message", "could not dial host", "host", host, "stack", microerror.Stack(err))
-				c.dialErrorCount.WithLabelValues(host).Inc()
-				return
-			}
-			defer conn.Close()
+		defer conn.Close()
 
-			elapsed := time.Since(start)
+		elapsed := time.Since(start)
 
-			c.latencyHistogramVec.Add(host, elapsed.Seconds())
-		}(host)
+		c.latencyHistogramVec.Add(host, elapsed.Seconds())
```

Comment:
could we also add a delay of `100ms` as Tuomas said in the sig infra sync? could be beneficial if this still don't fix the issue. Maybe we can go lower to `10ms` so this wont backfire on big clusters

Reply:
Could you remind me here the reason for that? I don't recall it. Which impact  has it?

Reply:
I've understood here something different since you were pointing to a different line. After reading twice, I got it what you mean. A delay in the loop. i think it makes sense.

Reply:
Added a small delay before the next loop but just after the calculation of Prometheus timing so we won't impact Prometheus metric